tower-http/src/sensitive_header.rs

```diff
@@ -0,0 +1,88 @@
+use futures_util::ready;
+use http::{header::HeaderName, Request, Response};
+use pin_project::pin_project;
+use std::future::Future;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_layer::Layer;
+use tower_service::Service;
+
+#[derive(Clone, Debug)]
+pub struct SensitiveHeaderLayer {
```

Comment:
question: should this be called SetSensitiveHeaderLayer? Might we have an UnsetSensitiveHeader?

Is it always appropriate to do this on both requests and responses? Should this layer hold a list of headers? Maybe a list for requests and a list for responses?

Reply:
I think renaming it to `SetSensitiveHeaderLayer` and `SetSensitiveHeader` is a good idea. I'll change that.

Not sure `UnsetSensitiveHeader` would be very useful as that is the default state and I don't quite see why you'd wanna make a header sensitive and then revert that later.

I think its also fine setting it on both the request and the response in one go. I'm thinking if some header is sensitive it'll be so regardless if its on request or response.

I actually thought setting it on the response is a bit odd, since I this is likely to be outer most middleware such that request headers are immediately marked as sensitive, before they reach logging/tracing. However if its the outer most one it'll also be the last to see the response which make marking response header less useful. I do think its slightly safer to keep it like this though.

Reply:
@olix0r wanna have another look?

Reply:
> Not sure `UnsetSensitiveHeader` would be very useful as that is the default state and I don't quite see why you'd wanna make a header sensitive and then revert that later.

I think that what @olix0r meant was a layer that _removes a header_ rather than a layer that unsets the sensitive flag on that header.

Reply:
> question: should this be called SetSensitiveHeaderLayer? Might we have an UnsetSensitiveHeader?
> 
> Is it always appropriate to do this on both requests and responses? Should this layer hold a list of headers? Maybe a list for requests and a list for responses?

IMO it might be worth having separate middleware for setting sensitive headers on requests and on responses? There could be a layer that produces each middleware, and a layer that produces both in one go...

Reply:
Ah yeah thats a good idea. I made some changes.